### PR TITLE
KP Overview

### PR DESCRIPTION
**Pull request by reynoldtan**
Implement KnowPulse overview and help page. KP overview page showcases many bioinformatics software/applications, use cases and benefits, whereas the Help page outlines many help topics and tutorials to assist users when using the portal.

Both pages can be access through the KnowPulse homepage.

More details and mock-ups in this issue.
Issue # https://github.com/UofS-Pulse-Binfo/kp_frontpage/issues/17

## Metadata

- [x] I feel this PR is ready to be merged.
- [ ] This PR is dependent upon [ ]
 
Documentation:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Description

1. Updated navigation bar to include link to overview page and help page.
2. Update module file to register menu items namely, overview, help and admin/update/help.
3. Implemented hook_preprocess_page() to suggest template.
4. Prepare additional template to accommodate page layout slightly different from overall theme.
5. Others (style sheet, JS to handle user events and behaviours, etc)

## Testing?

- [] I tested on a generic Tripal Site
- [X] I tested on a KnowPulse Clone
- [] This PR includes automated testing

### Installing kp_frontpage
1. Disable and Uninstall kp_frontpage module.
2. Update code base of kp_frontpage module.
3. Enable kp_frontpage module

### Updating help topics in Help page.
1. CMS to manage help topic items is available in host/admin/update/help
2. Prepare TSV file using prescribed column headers, namely. title, new, content_link, update_link. See table of column header described in CMS.
3. The order of items in the TSV file will also be the order with which help topics are rendered in the Help page.
4. The first 5 items in TSV file will be the first 5 topics shown in the Help page.

### Replacing video element
1. CMS is not provided to replace video.
2. In either page, overview or help, the same markup applies when embedding a video.
```
<div class="kp-video-container">
  <div>
    <div class="kp-video-presenter">
      <a href="#" class="kp-window-on kp-tip-top" data-text="Presented by: # NAME OF PRESENTER"><img src="<?php print $path_images . 'team/PRESENTER IMAGE.png'; ?>" height="90" width="90" /></a>
    </div>
      <iframe title="vimeo-player" src="YOUR LINK TO VIDEO HERE" height="450" frameborder="0" allowfullscreen></iframe>
  </div>
</div>
```

This markup will produce a video player with black screen background with the presenter image on top right corner.

3. Markup for either pages is located in templates/page.

NOTE: I highly recommend to just replace the scr attribute value of iframe in the markup above with the link to your video that is hosted in vimeo.

4. Change the data-text to the name of the presenter and image (img) src to match the photo of the presenter.

### CODE REVIEW

**Comment by ruobinLiu** [comment]
Looks great!

**Review by carolyncaron — COMMENTED** [review]
It looks really great Reynold, I have just some very minor suggestions!

### Overview Page
- "Bioinformatics" doesn't always fit on one line under the main header
Screen Shot 2020-12-11 at 4 10 28 PM

- In the pop-ups over each of our photos, it would be ideal to have some sort of separator between our roles and our department such as a comma or a new line. (For ex: "Bioinformaticist Pulse Crop Breeding and Genetics" reads awkwardly)
- The "Data Management" and "Project Management" title headings don't fit well on a condensed screensize (In my case, my window is occupying exactly half my 2880 x 1620 screen). Perhaps a solution for this is to change the font size dynamically with the window size? Or, like the help page, they can stack over each other instead of side-by-side at the same width.
Screen Shot 2020-12-11 at 3 33 48 PM

- There is a misspelling on the pop-up over Laura's photo in the video portion under Project Management. Should be: "Presented by: Laura Jardine"

### Help Page
- Under the title "Getting started with KnowPulse", the text reads "No matter which pulse crop data your are in" should be "No matter which pulse crop data you're interested in". Or maybe something even more succinct such as "No matter your pulse crop of interest"
- The abbreviation "KP" is used twice on this page - I'm worried some people won't understand. It shows up near the top as a link "KP Overview" and also in the paragraph before Laura's video. 
- In the paragraph under Ruobin's picture, her name is misspelled as Roubin. In the same paragraph, I think it would be clearer to include brackets or some delimiter behind the numbers (Ex. instead of "1 How to use pedigree tool..." something like "1) How to use pedigree tool..."
- I would change the sentence before Laura's video, where it starts with "Specifically,..." to "Specifically, Laura will demonstrate how KnowPulse was used for our two large-scale genomic projects - AGILE and EVOLVES." 
- For the crop linkouts, "Drybean" and "Fababean" should be "Dry Bean" and "Faba Bean" respectively.

I am using Firefox 76.0.1. Let me know if anything needs clarification!

**Comment by reynoldtan** [comment]
Implemented suggestions made by @carolyncaron 

This PR also fixes issues below:

Visible in the current homepage when:

Page is reduced to 90% and less
Screen Shot 2020-12-15 at 1 56 10 PM

Window is resized
Screen Shot 2020-12-15 at 1 58 29 PM

**Comment by reynoldtan** [comment]
Screen Shot 2020-12-15 at 2 18 57 PM

**Review by laceysanderson — APPROVED** [review]
## Functional Review
I did a very quick check of the functionality and it looks good 👍 

## Code Review
 - ✅ I like that the CSS is very specific
 - ✅ JS is well commented and variable names are readable
 - ✅ JS logic looks good
 - ⚠️ Particle JS should have been included using libraries not directly in the repository.
 - ⚠️ copy/pasted comment in hook_menu from phenotypes ;-)
 - ✅ help topic cms code in kp_frontpage.module looks good
 - ✅ page preprocess looks ok
 - ✅ Blank template page html: looks good
 - ✅ Help.php template: looks good
 - ✅ Overview.php template: looks good
 - ⚠️ all html pages should be wrapped to ~80 characters to match coding standards

Regarding Particle.js and the other warnings: they can be fixed in a separate PR after this is made live.

File: templates/page--kppage.tpl.php [hunk 1]
```diff
@@ -22,23 +22,23 @@
...
+      <br /><a href="https://www.rdc-drc.ca/research-data-canada-endorses-trust-principles-for-trustworthy-digital-data-repositories" target="_blank">www.rdc-drc.ca</a>
       <br /><br />
     </li>
     <li>
       <strong>The Breeding API (BrAPI)</strong> project is an effort to enable interoperability
       among plant breeding databases. BrAPI is a standardized RESTful web service API specification
       for communicating plant breeding data. This community driven standard is free to be used by
       anyone interested in plant breeding data management.
-      <br /><a href="http://opensource.org" target="_blank">www.opensource.org</a>
+      <br /><a href="https://brapi.docs.apiary.io" target="_blank">www.brapi.docs.apiary.io</a>
```
Comment (laceysanderson) [inline templates/page--kppage.tpl.php]: Should be https://brapi.org/ not the apiary

**Pull request merged**